### DISCUSSION TITLE
Build Ubuntu 22.04 with Rockchip Linux 5.10

### DISCUSSION THREAD

**Discussion by wmport** [Joshua-Riek/ubuntu-rockchip#682] [General]
Build Ubuntu 22.04 with Rockchip Linux 5.10 #328
nanopc-t6

drm not work

mpv ffmpeglist 

Playing: http://10.0.10.125:2022/E442C20D
 (+) Video --vid=1 (h264 1920x1080 25.000fps)
 (+) Audio --aid=1 (ac3 2ch 48000Hz)
[ffmpeg/video] h264_rkmpp: Failed to init MPP context: -1
Could not open codec.
[ffmpeg] Impossible to convert between the formats supported by the filter 'mpv_src_default_in' and the filter 'auto_scale_0'
[lavfi] failed to configure the filter graph
Disabling filter scale_rkrga.00 because it has failed.
AO: [pulse] 48000Hz stereo 2ch float
VO: [gpu] 1920x1080 yuv420p
AV: 00:00:33 / 00:00:49 (66%) A-V:  0.000 Dropped: 31 Cache: 16s/18MB

Exiting... (Quit)

ubuntu@ubuntu-desktop:~/Videos$ mpv HDR10-Life-Untouched-\(www.demolandia.net\).mp4 
 (+) Video --vid=1 (*) (hevc 3840x2160 59.940fps)
 (+) Audio --aid=1 (*) (aac 2ch 48000Hz)
[ffmpeg/video] hevc_rkmpp: Failed to init MPP context: -1
Could not open codec.
[ffmpeg] Impossible to convert between the formats supported by the filter 'mpv_src_default_in' and the filter 'auto_scale_0'
[lavfi] failed to configure the filter graph
Disabling filter scale_rkrga.00 because it has failed.
AO: [pulse] 48000Hz stereo 2ch float
VO: [gpu] 3840x2160 yuv420p10
AV: 00:00:00 / 00:03:17 (0%) A-V:  0.399 Dropped: 13

Audio/Video desynchronisation detected! Possible reasons include too slow
hardware, temporary CPU spikes, broken drivers, and broken files. Audio
position will not match to the video (see A-V status field).

AV: 00:00:05 / 00:03:17 (3%) A-V:  9.716 Dropped: 228

Exiting... (Quit)
ubuntu@ubuntu-desktop:~/Videos$ 

in logs

14:52:02 mpv: mpp_mem_pool: put_pool found 4 used buffer size 224
14:52:02 mpv: mpp_mem_pool: put_pool found 4 used buffer size 224
14:51:46 mpv: mpp: error found on mpp initialization
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:51:46 mpv: mpp_dec: mpp_dec_init could not init hal
14:51:46 mpv: mpp_hal: mpp_hal_init could not found coding type 16777220
14:51:46 mpv: mpp_hal: mpp_hal_init hal h265d_rkdec init failed ret -1
14:51:46 mpv: hal_h265d_api: mpp_dev_init failed ret: -1
14:51:46 mpv: vcodec_service: open vcodec_service /dev/mpp_service failed
14:49:02 mpv: mpp_mem_pool: put_pool found 4 used buffer size 224
14:48:29 mpv: mpp: error found on mpp initialization
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_time: mpp_clock_put invalid clock (nil)
14:48:29 mpv: mpp_dec: mpp_dec_init could not init hal
14:48:29 mpv: mpp_hal: mpp_hal_init could not found coding type 7
14:48:29 mpv: mpp_hal: mpp_hal_init hal h264d_rkdec init failed ret -1
14:48:29 mpv: hal_h264d_api: mpp_dev_init failed ret: -1
14:48:29 mpv: vcodec_service: open vcodec_service /dev/mpp_service failed
14:47:19 kernel: mpp_rkvdec2 fdc48100.rkvdec-core: shared_niu_h is not found!
14:47:19 kernel: mpp_rkvdec2 fdc48100.rkvdec-core: shared_niu_h is not found!
14:47:19 kernel: mpp_rkvdec2 fdc48100.rkvdec-core: shared_niu_a is not found!
14:47:19 kernel: mpp_rkvdec2 fdc38100.rkvdec-core: shared_niu_h is not found!
14:47:19 kernel: mpp_rkvdec2 fdc38100.rkvdec-core: shared_niu_a is not found!
14:47:19 kernel: mpp-iep2 fdbb0000.iep: allocate roi buffer failed

ubuntu@ubuntu-desktop:~$ ffmpeg 
ffmpeg version 6.0.1-1+git20240229.99d938d6~jammy Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 11 (Ubuntu 11.4.0-1ubuntu1~22.04)
  configuration: --prefix=/usr --extra-version='1+git20240229.99d938d6~jammy' --toolchain=hardened --libdir=/usr/lib/aarch64-linux-gnu --incdir=/usr/include/aarch64-linux-gnu --arch=arm64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-version3 --enable-rkmpp --enable-sdl2 --enable-rkrga --disable-sndio --enable-pocketsphinx --enable-librsvg --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-libplacebo --enable-shared
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

Use -h to get full help or, even better, run 'man ffmpeg'
ubuntu@ubuntu-desktop:~$

**Comment by Joshua-Riek** [comment]
Ah, I broke the default user groups. The user account must be apart of a few pre-defined user groups.
```
ubuntu@ubuntu-desktop:~$ groups
ubuntu adm cdrom sudo dip plugdev sambashare lpadmin
```
We can see here that EXTRA_GROUPS is broken, I likely have a bug in my sed statement.
```
ubuntu@ubuntu-desktop:~/Desktop$ cat /etc/adduser.conf 
# /etc/adduser.conf: `adduser' configuration.
# See adduser(8) and adduser.conf(5) for full documentation.

# The DSHELL variable specifies the default login shell on your
# system.
DSHELL=/bin/bash

# The DHOME variable specifies the directory containing users' home
# directories.
DHOME=/home

# If GROUPHOMES is "yes", then the home directories will be created as
# /home/groupname/user.
GROUPHOMES=no

# If LETTERHOMES is "yes", then the created home directories will have
# an extra directory - the first letter of the user name. For example:
# /home/u/user.
LETTERHOMES=no

# The SKEL variable specifies the directory containing "skeletal" user
# files; in other words, files such as a sample .profile that will be
# copied to the new user's home directory when it is created.
SKEL=/etc/skel

# FIRST_SYSTEM_[GU]ID to LAST_SYSTEM_[GU]ID inclusive is the range for UIDs
# for dynamically allocated administrative and system accounts/groups.
# Please note that system software, such as the users allocated by the base-passwd
# package, may assume that UIDs less than 100 are unallocated.
FIRST_SYSTEM_UID=100
LAST_SYSTEM_UID=999

FIRST_SYSTEM_GID=100
LAST_SYSTEM_GID=999

# FIRST_[GU]ID to LAST_[GU]ID inclusive is the range of UIDs of dynamically
# allocated user accounts/groups.
FIRST_UID=1000
LAST_UID=59999

FIRST_GID=1000
LAST_GID=59999

# The USERGROUPS variable can be either "yes" or "no".  If "yes" each
# created user will be given their own group to use as a default.  If
# "no", each created user will be placed in the group whose gid is
# USERS_GID (see below).
USERGROUPS=yes

# If USERGROUPS is "no", then USERS_GID should be the GID of the group
# `users' (or the equivalent group) on your system.
USERS_GID=100

# If DIR_MODE is set, directories will be created with the specified
# mode. Otherwise the default mode 0755 will be used.
DIR_MODE=0750

# If SETGID_HOME is "yes" home directories for users with their own
# group the setgid bit will be set. This was the default for
# versions << 3.13 of adduser. Because it has some bad side effects we
# no longer do this per default. If you want it nevertheless you can
# still set it here.
SETGID_HOME=no

# If QUOTAUSER is set, a default quota will be set from that user with
# `edquota -p QUOTAUSER newuser'
QUOTAUSER=""

# If SKEL_IGNORE_REGEX is set, adduser will ignore files matching this
# regular expression when creating a new home directory
SKEL_IGNORE_REGEX="dpkg-(old|new|dist|save)"

# Set this if you want the --add_extra_groups option to adduser to add
# new users to other groups.
# This is the list of groups that new non-system users will be added to
# Default:
EXTRA_GROUPS="video adm dialout cdrom audio plugdev netdev input bluetooth""dialout cdrom floppy audio video plugdev users"

# If ADD_EXTRA_GROUPS is set to something non-zero, the EXTRA_GROUPS
# option above will be default behavior for adding new, non-system users
#ADD_EXTRA_GROUPS=1


# check user and group names also against this regular expression.
#NAME_REGEX="^[a-z][-a-z0-9_]*\$"

# use extrausers by default
#USE_EXTRAUSERS=1
```
